Support ceil_mode=True on maxpool2d in XNNPACK delegate

GregoryComer opened the pull request:
Summary: Support ceil_mode=True for MaxPool2d in XNNPACK delegate by conditionally updating padding. This works only when the input shape is static, as it requires statically computing the change in padding. As such, the partitioner constraints are also updated to reflect this.

Differential Revision: D67386151

mcr229 reviewed (approved):
This is smart, i like it

GregoryComer committed:
Support ceil_mode=True on maxpool2d in XNNPACK delegate (#7355)

Summary:
Support ceil_mode=True for MaxPool2d in XNNPACK delegate by conditionally updating padding. This works only when the input shape is static, as it requires statically computing the change in padding. As such, the partitioner constraints are also updated to reflect this.


Test Plan: Added a new test (test_fp32_maxpool2d_ceilmode) that exercises various permutations of input with ceil_mode=True.

Reviewed By: digantdesai, mcr229

Differential Revision: D67386151

Pulled By: GregoryComer
